### PR TITLE
Fix star hover state

### LINKED ISSUE #985: stop hover state once the user clicks on a star
# STR

1. Hover over a star and see the star color change
2. Now click it

# Expected

For the star to show it's actual state, whether that be checked or not checked, while the user is still hovered over it.

# Actual

The star continues to show the hover state while the user is hovered over the icon they just clicked.

### LINKED ISSUE #986: Add outline to star (unstarred state) for selected source
**EDIT:** Issue has significantly evolved. Scroll down to last comment, to see.

---
_Addendum to #873_ 

---

## Problem
When a user goes to click on a star to star or un-star a Source, the hover is triggered first—yay! Unfortunately, when the user then clicks to star or un-star a Source, they do not receive immediate feedback on that action—as they continue to see the over state via their cursor still hovering over the star. 

To not be confusing, I feel immediate and visually demonstrable feedback should happen, when a user clicks on anything—even when the cursor is still hovering over that item.

## Solution
The GMail team has solved for this nicely. They also user-test the hell out of everything, so I trust their implemented solutions as going over with users, well. In the GIF below, the following pattern is demonstrated (using language we've coined for our own project, to speak to the GMail UI):
1. When a user hovers over the entire Source widget, that full-widget hover state is demonstrated by a thickened/darkened outline on the Star, a thickened/darkened outline on the Checkbox, and grabber dots to the far left. 
2. On `hover` the star shows both the darker/thickened outline, and a gray circle behind it.
3. On `click`...
    3a. When the mouse button is released from the click action, the grey circle flashes away, then animates in size from small to large, all in a half-second. The GIF shows a solid circle outline appear, but that's my screencap software demonstrating the mouse click.
    3b. The star immediately reflects the state-change, despite the user's cursor still being over the star. To reflect this maintained hover, the circle also remains behind the star.
4. When the user mouses away/off from the star, the circle goes away and the on/off toggle remains in the state the user just acted upon.
5. Tooltips are also shown for "starred" and "unstarred" upon hover-over. Eventually these would be great, but feel very optional considering other priorities and resource limitations.
ezgif com-video-to-gif(1)

I do not wish to act upon item #1, in part because we have neither a grabber affordance nor checkboxes. 3a also feels too fancy. For this ticket, however, I would like to act on items 2, 3b, and 4. IGNORE the numbers on the below mockup, they correspond to a different issue.

image

Artwork to do this, will be pasted below as SVG code.

### PR DESCRIPTION
**Pull request by ntoll**
# Description

Fixes #985
Towards #986

This PR fixes the above two issues to the extent that Erik has described the expected initial solution (todo: hover background).

Screenie here:

starmageddon

# Test Plan

Eyeball Mk.1 :eyes:

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:

 - [ ] I have tested these changes in the appropriate Qubes environment
 - [ ] I do not have an appropriate Qubes OS workstation set up (the reviewer will need to test these changes)
 - [X] These changes should not need testing in Qubes

If these changes add or remove files other than client code, packaging logic (e.g., the AppArmor profile) may need to be updated. Please check as applicable:

 - [ ] I have submitted a separate PR to the [packaging repo](https://github.com/freedomofpress/securedrop-debian-packaging)
 - [X] No update to the packaging logic (e.g., AppArmor profile) is required for these changes
 - [ ] I don't know and would appreciate guidance

### CODE REVIEW

**Comment by eloquence** [comment]
Let's keep #986 open for now and only close #985 with this PR.

**Comment by eloquence** [comment]
(Taking this for a quick spin.)

**Comment by eloquence** [comment]
:+1: Works as advertised and improves current starring behavior. (Only tested in non-Qubes for now.)

File: securedrop_client/gui/widgets.py [hunk 1]
```diff
@@ -1351,7 +1351,7 @@ def eventFilter(self, obj, event):
         t = event.type()
         if t == QEvent.HoverEnter:
             self.setIcon(load_icon('star_hover.svg'))
-        elif t == QEvent.HoverLeave:
+        elif t == QEvent.HoverLeave or t == QEvent.MouseButtonPress:
```
Comment (creviera) [inline securedrop_client/gui/widgets.py]: makes sense :+1:

**Comment by creviera** [comment]
just a thought for a future improvement: if the star is already checked, then don't show the hover state when you mouse over it, otherwise it looks like a star that is unchecked about to be checked

**Pull request merged**